src/components/hooks/use-user-browser.tsx

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from 'react'
+import { browserName, browserVersion } from 'react-device-detect'
+
+export const useUserBrowser = () => {
+    const [browser, setBrowser] = useState({
+        browser_name: '',
+        browser_version: 0,
+    })
+    useEffect(() => {
+        setBrowser({
+            browser_name: browserName,
+            browser_version: Number(browserVersion),
+        })
+    }, [])
+
+    return browser
```

Comment:
```ts
import { useEffect, useState } from 'react'
import { browserName, browserVersion } from 'react-device-detect'

const minimum_required_versions = {
    Chrome: 115,
    Firefox: 116,
    Opera: 107,
    Yandex: 108,
    Safari: 109,
    'Internet Explorer': 110,
    Edge: 111,
    Chromium: 112,
    IE: 113,
    'Mobile Safari': 114,
    'Edge Chromium': 115,
    'MIUI Browser': 116,
    'Samsung Browser': 117,
}

export const useUserBrowser = () => {
    const [browser, setBrowser] = useState({
        browser_name: '',
        browser_version: 0,
    })
    const is_outdated = minimum_required_versions[browser_name] &&
            browser_version < minimum_required_versions[browser_name];

    useEffect(() => {
        setBrowser({
            browser_name: browserName,
            browser_version: Number(browserVersion),
        })
    }, [])

    return {
        ...browser,
        is_outdated,
    }
}
```

@iman-fs You can move your logic inside the hook itself, So you won’t need to rewrite it again in the future. By doing so you can also move the `minimum_required_versions` to the hook itself and isolate it, There is no reason for jumping between two files to understand the code here.

By the way, the above code is just an example 😅 Modify with your needs 🙇🏻

Reply:
you're right. I put the constants in the hook at first but thought maybe it's not a good idea. will refactor the code. thanks for your revision :pray:

Reply:
@farzin-deriv the code is updated now. I could remove browser information from the return part of the hook but I kept it there. maybe later we can use it somewhere else because the whole purpose of this hook is to give the user browser information

Reply:
Nice work man 🚀